### ISSUE TITLE
error installing on a docker env

### ISSUE THREAD

**Issue by daviddoji** [pygraphviz/pygraphviz#201]
I'm getting the following error when trying to install the library in a docker environment.

```
Installing collected packages: pygraphviz
  Running setup.py install for pygraphviz: started
    Running setup.py install for pygraphviz: finished with status 'error'
/opt/conda/lib/python3.7/site-packages/pip/_internal/commands/install.py:244: UserWarning: Disabling all use of wheels due to the use of --build-options / --global-options / --install-options.
  cmdoptions.check_install_build_global(options)
    ERROR: Complete output from command /opt/conda/bin/python -u -c 'import setuptools, tokenize;__file__='"'"'/tmp/pip-install-qw5erzsy/pygraphviz/setup.py'"'"';f=getattr(tokenize, '"'"'open'"'"', open)(__file__);code=f.read().replace('"'"'\r\n'"'"', '"'"'\n'"'"');f.close();exec(compile(code, __file__, '"'"'exec'"'"'))' install --record /tmp/pip-record-ruw4rfx5/install-record.txt --single-version-externally-managed --compile --include-path=/opt/conda/lib/python3.7/site-packages/graphviz --library-path=/opt/conda/lib/python3.7/site-packages/graphviz/lib:
    ERROR: running install
    running build
    running build_py
    creating build
    creating build/lib.linux-x86_64-3.7
    creating build/lib.linux-x86_64-3.7/pygraphviz
    copying pygraphviz/release.py -> build/lib.linux-x86_64-3.7/pygraphviz
    copying pygraphviz/graphviz.py -> build/lib.linux-x86_64-3.7/pygraphviz
    copying pygraphviz/__init__.py -> build/lib.linux-x86_64-3.7/pygraphviz
    copying pygraphviz/agraph.py -> build/lib.linux-x86_64-3.7/pygraphviz
    copying pygraphviz/version.py -> build/lib.linux-x86_64-3.7/pygraphviz
    creating build/lib.linux-x86_64-3.7/pygraphviz/tests
    copying pygraphviz/tests/test_clear.py -> build/lib.linux-x86_64-3.7/pygraphviz/tests
    copying pygraphviz/tests/test_edge_attributes.py -> build/lib.linux-x86_64-3.7/pygraphviz/tests
    copying pygraphviz/tests/test_setup.py -> build/lib.linux-x86_64-3.7/pygraphviz/tests
    copying pygraphviz/tests/test_node_attributes.py -> build/lib.linux-x86_64-3.7/pygraphviz/tests
    copying pygraphviz/tests/test.py -> build/lib.linux-x86_64-3.7/pygraphviz/tests
    copying pygraphviz/tests/test_readwrite.py -> build/lib.linux-x86_64-3.7/pygraphviz/tests
    copying pygraphviz/tests/test_subgraph.py -> build/lib.linux-x86_64-3.7/pygraphviz/tests
    copying pygraphviz/tests/__init__.py -> build/lib.linux-x86_64-3.7/pygraphviz/tests
    copying pygraphviz/tests/test_graph.py -> build/lib.linux-x86_64-3.7/pygraphviz/tests
    copying pygraphviz/tests/test_string.py -> build/lib.linux-x86_64-3.7/pygraphviz/tests
    copying pygraphviz/tests/test_unicode.py -> build/lib.linux-x86_64-3.7/pygraphviz/tests
    copying pygraphviz/tests/test_layout.py -> build/lib.linux-x86_64-3.7/pygraphviz/tests
    copying pygraphviz/tests/test_html.py -> build/lib.linux-x86_64-3.7/pygraphviz/tests
    copying pygraphviz/tests/test_attributes.py -> build/lib.linux-x86_64-3.7/pygraphviz/tests
    copying pygraphviz/tests/test_drawing.py -> build/lib.linux-x86_64-3.7/pygraphviz/tests
    copying pygraphviz/tests/test_attribute_defaults.py -> build/lib.linux-x86_64-3.7/pygraphviz/tests
    running egg_info
    writing pygraphviz.egg-info/PKG-INFO
    writing dependency_links to pygraphviz.egg-info/dependency_links.txt
    writing top-level names to pygraphviz.egg-info/top_level.txt
    reading manifest file 'pygraphviz.egg-info/SOURCES.txt'
    reading manifest template 'MANIFEST.in'
    warning: no files found matching '*.png' under directory 'doc'
    warning: no files found matching '*.html' under directory 'doc'
    warning: no files found matching '*.txt' under directory 'doc'
    warning: no files found matching '*.css' under directory 'doc'
    warning: no previously-included files matching '*~' found anywhere in distribution
    warning: no previously-included files matching '*.pyc' found anywhere in distribution
    warning: no previously-included files matching '.svn' found anywhere in distribution
    no previously-included directories found matching 'doc/build'
    writing manifest file 'pygraphviz.egg-info/SOURCES.txt'
    copying pygraphviz/graphviz.i -> build/lib.linux-x86_64-3.7/pygraphviz
    copying pygraphviz/graphviz_wrap.c -> build/lib.linux-x86_64-3.7/pygraphviz
    running build_ext
    building 'pygraphviz._graphviz' extension
    creating build/temp.linux-x86_64-3.7
    creating build/temp.linux-x86_64-3.7/pygraphviz
    gcc -pthread -B /opt/conda/compiler_compat -Wl,--sysroot=/ -Wsign-compare -DNDEBUG -g -fwrapv -O3 -Wall -Wstrict-prototypes -fPIC -I/opt/conda/lib/python3.7/site-packages/graphviz -I/opt/conda/include/python3.7m -c pygraphviz/graphviz_wrap.c -o build/temp.linux-x86_64-3.7/pygraphviz/graphviz_wrap.o
    pygraphviz/graphviz_wrap.c:2987:10: fatal error: graphviz/cgraph.h: No such file or directory
     #include "graphviz/cgraph.h"
              ^~~~~~~~~~~~~~~~~~~
    compilation terminated.
    error: command 'gcc' failed with exit status 1
    ----------------------------------------
ERROR: Command "/opt/conda/bin/python -u -c 'import setuptools, tokenize;__file__='"'"'/tmp/pip-install-qw5erzsy/pygraphviz/setup.py'"'"';f=getattr(tokenize, '"'"'open'"'"', open)(__file__);code=f.read().replace('"'"'\r\n'"'"', '"'"'\n'"'"');f.close();exec(compile(code, __file__, '"'"'exec'"'"'))' install --record /tmp/pip-record-ruw4rfx5/install-record.txt --single-version-externally-managed --compile --include-path=/opt/conda/lib/python3.7/site-packages/graphviz --library-path=/opt/conda/lib/python3.7/site-packages/graphviz/lib" failed with error code 1 in /tmp/pip-install-qw5erzsy/pygraphviz/
```

The library graphviz is already installed:
`Requirement already satisfied: graphviz in /opt/conda/lib/python3.7/site-packages (0.11.1)`

I'm using this command for the installation:
`pip install pygraphviz --install-option="--include-path=/opt/conda/lib/python3.7/site-packages/graphviz" --install-option="--library-path=/opt/conda/lib/python3.7/site-packages/graphviz/lib"`

Any idea how to solve it?

Thanks in advance

```

**Comment by tachyontraveler** [comment]
The same error happens with my pip installation on Ubuntu 20.04 and Python 3.7.9 inside a conda environment. 
`pygraphviz/graphviz_wrap.c:2676:10: fatal error: graphviz/cgraph.h: No such file or directory`
   `2676 | #include "graphviz/cgraph.h"`

Eventually, I installed pygraphviz using `conda install`, but the corresponding version was v1.3 (current version in pip when I'm writing this reply is v1.6)

It'd be great if anyone could solve this issue. Thanks!

**Comment by jarrodmillman** [comment]
I created two new issues:
- https://github.com/conda-forge/graphviz-feedstock/issues/55
- https://github.com/conda-forge/pygraphviz-feedstock/issues/13

And am planning to try to follow-up with anaconda once conda-forge updates things:
- https://anaconda.org/anaconda/graphviz
- https://anaconda.org/anaconda/pygraphviz

It would be great if you could comment and / or thumbs up the conda-forge issues I created.  Hopefully, we can get things fixed soon.

**Comment by rossbar** [comment]
There was an issue with custom setup extensions in pygraphviz that were causing problems with finding headers/libs on Ubuntu 20.04 - this has been fixed in pygraphviz-1.7. The out-of-date packages on conda is a separate issue - please see the links above. 

If you have Graphviz installed via the [platform package manager (e.g. `apt` in Ubuntu)](https://pygraphviz.github.io/documentation/stable/install.html#ubuntu-and-debian), pygraphviz-1.7 should be pip installable. If you were using `conda` to manage different versions of Graphviz, you will likely still have problems with out-of-date packages.

Note that you should be able to install pygraphviz-1.7 via pip *within a conda environment* on Ubuntu as long as Graphviz is installed, e.g.

```
sudo apt install graphviz graphviz-dev
conda create -n testenv
conda activate testenv
pip install pygraphviz==1.7
```

I'm going to close this for now - please feel free to reopen if problems persist installing the latest pygraphviz on Ubuntu.
